.github/workflows/int-and-connected-test-run.yml

```diff
@@ -1,153 +1,85 @@
     steps:
-      - name: "Checkout code"
-        uses: actions/checkout@v3
-      - name: "Cache build"
-        uses: actions/cache@v3
-        with:
-          path: ${{ env.CACHE_PATHS }}
-          key: ${{ runner.os }}-build-connected
-      - name: "Write RBS SA to a file"
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin'
+          java-version: '17'
+          cache: 'gradle'
+      - name: Run connected tests
+        env:
+          # required for azure tests
+          AZURE_CREDENTIALS_APPLICATIONID: 0e29ec36-04e8-44d5-ae7c-50dc15135571
+          AZURE_CREDENTIALS_HOMETENANTID: fad90753-2022-4456-9b0a-c7e5b934e408
+          AZURE_CREDENTIALS_SECRET: ${{ secrets.AZURE_CREDENTIALS_SECRET }}
+          # required for synapse tests
+          AZURE_SYNAPSE_SQLADMINPASSWORD: ${{ secrets.AZURE_SYNAPSE_SQLADMINPASSWORD }}
+          AZURE_SYNAPSE_SQLADMINUSER: ${{ secrets.AZURE_SYNAPSE_SQLADMINUSER }}
+          AZURE_SYNAPSE_WORKSPACENAME: tdr-snps-int-east-us-ondemand.sql.azuresynapse.net
+          # required for connected tests
+          GOOGLE_APPLICATION_CREDENTIALS: jade-dev-account.json
+          # required data project for snapshotTest
+          GOOGLE_CLOUD_DATA_PROJECT: broad-jade-integration-data
+          # required for testAzureBillingProfile
+          # uses an azure marketplace app with this hardcoded deployment email
+          JADE_USER_EMAIL: connected-tdr-user@notarealemail.org
+          # required for rbs tests
+          RBS_CLIENT_CREDENTIAL_FILE_PATH: rbs-tools-sa.json
+          # output plain logs instead of json
+          TDR_LOG_APPENDER: 'Console-Standard'
         run: |
-          # write vault token
+          # extract service account credentials
+          base64 --decode <<< ${{ secrets.SA_B64_CREDENTIALS }} > ${GOOGLE_APPLICATION_CREDENTIALS}
           base64 --decode <<< ${{ secrets.B64_RBS_APPLICATION_CREDENTIALS }} > ${RBS_CLIENT_CREDENTIAL_FILE_PATH}
-      - name: "Run connected tests via Gradle"
-        uses: broadinstitute/datarepo-actions/actions/main@0.74.0
-        with:
-          actions_subcommand: 'gradleinttest'
-          pgport: ${{ job.services.postgres.ports[5432] }}
-          test_to_run: 'testConnected'
-          sa_b64_credentials: ${{ secrets.SA_B64_CREDENTIALS }}
-      - name: "Temp: Archive all junit test reports"
-        uses: actions/upload-artifact@v2
-        if: always()
-        with:
-          name: junit-test-reports-for-connected
-          path: build/reports
-          retention-days: 5
-  deploy_test_integration:
-    name: "Run integration and smoke tests"
-    outputs:
-      job-status: ${{ job.status }}
-      api_image_tag: ${{ steps.configuration.outputs.git_hash }}
+          # assemble code and run connected tests
+          ./gradlew assemble
```

Comment:
Did the old workflows not use a build cache for gradle (to share build outputs between jobs)? Something to think about for a future PR; I have't used https://github.com/gradle/actions/blob/main/docs/setup-gradle.md but it looks interesting.

Reply:
The old workflows use the `cache` action, instead of the one built into `actions/setup-java@v4`. The java one is more robust and doesn't require a cache key.